unity-renderer/Assets/DCLServices/CameraReelService/CameraReelNetworkStorageService.cs

```diff
@@ -11,6 +11,8 @@ public class CameraReelNetworkStorageService : ICameraReelStorageService
 
         private ICameraReelAnalyticsService analyticsServiceLazy;
 
+        public event Action<Texture2D, ScreenshotMetadata, UniTaskCompletionSource<(CameraReelResponse, CameraReelStorageStatus)>> Upload;
```

Comment:
Initial idea here is that we can temporarily put picture in the Gallery before uploading is finished (ASAP), because we have all the data needed. Then, once we get a result of uploading - success or error, we either leave the entry in the gallery, or we remove it if it is failed. Additionally, we won't need to download picture from the web reducing performance and cost. However, maybe this mechanism would imply bigger refactor on the Gallery side

Reply:
As we agreed on our sync discussion and following KISS and YAGNI, I will go to have only one `ScreenshotUploaded `event with the `CameraReelResponse, CameraReelStorageStatus` only